models/reviews.js

```diff
@@ -33,6 +34,26 @@ exports.fetchReviewById = (review_id) => {
     })
 }
 
+exports.fetchCommentsByReviewId = (review_id) => {
+
+    return this.fetchReviewById(review_id)
+    .then(() => {
+        return db.query(`
+        SELECT comments.* FROM reviews
+        JOIN comments ON reviews.review_id = comments.review_id
+        WHERE reviews.review_id = $1;`, [review_id])
+        .then(({ rows: comments }) => {
+            if (!comments.length){
+                return 'No Comments!';
+            }
+            return comments;
+        })
+    })
+    .catch((err) => {
```

Comment:
If there's an error, this will be passed up to the controller where it will be caught in our middleware, so there's no need to reject -> purely by the fact that there's already an error to reference (on line 52) means we don't have to cause one by rejecting the promise.

Reply:
Thanks for the confirmation on this. I may end up refactoring my entire error handling once I'm ready to push it to v1.0, but this is a good point. Appreciate the explanation and clarification, will clear this up for the next PR!